src/components/Home.js

```diff
@@ -1,20 +1,70 @@
-import React from 'react';
-import { NavLink } from 'react-router-dom';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useCollectionData } from 'react-firebase-hooks/firestore';
+import firebase from '../lib/firebase';
 import getToken from '../lib/tokens';
+import '../styles/Home.css';
+
+const db = firebase.firestore().collection('shopping_list');
 
 const Home = () => {
+  const [existingToken, setExistingToken] = useState('');
+  const history = useHistory();
+
   const newList = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+    history.push('/list');
+  };
+
+  const tokenHandler = (event) => {
+    setExistingToken(event.target.value);
   };
+
+  const [shoppingList] = useCollectionData(
+    db.where('token', '==', existingToken),
+  );
+
+  const submitToken = (e) => {
+    e.preventDefault();
+
+    if (existingToken === '') {
+      alert('Please enter a token...');
+      return;
+    }
+
+    if (shoppingList.length) {
+      localStorage.setItem('token', existingToken);
+      history.push('/list');
+    } else {
+      alert('Token does not exist! Please try again or create a new list.');
+    }
+  };
+
   return (
     <div>
-      <h1>Welcome to Smart Shopping App</h1>
-      <NavLink to="/list">
-        <button type="submit" onClick={newList}>
-          Create List
-        </button>
-      </NavLink>
+      <div className="home">
+        <div>
+          <h1>Welcome to Smart Shopping App</h1>
+          <button type="submit" onClick={newList}>
+            Create List
+          </button>
+        </div>
+        <div>
+          <p>- or -</p>
```

Comment:
Just a question for my understanding! What does this line do?

Reply:
The `<p>- or -</p>` is just a single line for styling (like a separator for the two buttons) to divide the two option you have on the homepage.

Reply:
Ah ok, thanks for explaining - I haven't seen that before!